src/common/types.ts

```diff
@@ -37,6 +37,8 @@ export type DictionarySourceType =
  * A word list stored in the database.
  */
 export interface WordListSource {
+  /** Autogenerated primary key */
+  id?: number;
```

Comment:
Thanks for bringing it up! I haven't thought about this, I looked into the code and couldn't find a use case for this one just because the `updateManualEntryDictionaryToProject` which explicitly pass in `id` would eventually call `saveFile` which takes in `StoredWordList` which has `id` attribute as optional. Though I might be mistaken, could you elaborate more on this idea?

Reply:
I think I see where you're coming from! The interface I suggested is just a duplicate of `StoredWordList`, so feel free to disregard that suggestion

To elaborate, my concern with adding optional attributes to `WordListSource` is that there are fewer guarantees about what the object contains. Since `WordListSource` is used in a few areas, there might be times where the object doesn't have an `id` property. This might make it harder to reason about whether or not `id` is available at a certain point of code

I might be misreading, but I think your comment is addressing my concern by stating that there are very few times where `id` will be null.

What are your thoughts on removing the optional `id` property on `WordListSource`, and converting all variables that need the `id` property into type `StoredWordList`? For example, what if we convert `sources` in `LanguageSources.svelte` to `StoredWordList[]` and `tableData` in `ManualEntry.svelte` to `StoredWordList`? This way there might be more clarity as to whether or not the object contains the `id` property

Reply:
Good point! @eddieantonio I was wondering if you have any comments/suggestions on this topic.

Reply:
I agree with @ChrisChrisLoLo. Optional properties are icky, haha! And we _know_ that a wordlist fetched from the database has a valid `id` property. In many other areas of the codebase, the `id` property does not matter, and so they can take a `WordListSource` _without_ the `id` property!

Reply:
I've spent some time investigating this issue, and I think we would have to implement a separate workflow/component to deal with adding a new direct entry vs. editing an existing one. The issue is mainly coming from the fact that they're sharing `tableData` which is currently set to `WordListSource` i.e. missing an `id` attr. If we would like to use `IndexedWordListSource` to enforce having an `id` attribute, then those that are missing an `id` would have a filler value as `id` which also needs to be handled on the worker side.

I might be mistaken, if you have any suggestions as to how to introduce `IndexedWordListSource`, I'd really appreciate it!

Reply:
An straightforward way of dealing with this is to create an _empty_ direct entry wordlist, store it, and _then_ modify it. This will be entirely transparent in the UI, but this avoids creating a different workflow for a "new" direct entry wordlist and an existing one; there will _always_ be an existing direct entry wordlist; the only difference is that a new, empty wordlist is created automatically.

If this becomes too involved, I have no problems with spinning this off into its own issue that can be dealt separately outside of this PR.

Reply:
@eddieantonio Thanks for your insight! I've had a similar idea to that as in when the user switches to a direct entry tab, we would create an empty entry (we then would have access to its `id`) to be modified further. I was worried it would either create a new entry every time the `AddSource` re-renders or we would need to check if there's an empty entry created already for this purpose. I'll create a separate issue referencing this thread to investigate further!